### PR TITLE
fix(logs): honor klog logging options

### LINKED ISSUE #122: Do not create log_file
**Describe the bug:** Do not create log_file

**Expected behaviour:** log_file can created at the right location

**Steps to reproduce the bug:**
1.set args --logtostrerr=false and --log_file=/var/log/localpv.log
2.start container

**The output of the following commands will help us better understand what's going on**:

* get into container and to see not the "localpv.log" file created at /var/log/``

**Anything else we need to know?:**
maybe vendor/github.com/openebs/maya/pkg/logs/logs.go using the klog v1 NOT v2

**Environment details:**
- OpenEBS version v3.0.1

### PR DESCRIPTION
**Pull request by niladrih**
**Why is this PR required? What issue does it fix?**:
Fixes #122 
This makes changes so that the user defined klog flags are honored

**What this PR does?**:
1. Uses and parses pflag.Commandline FlagSet as a central global FlagSet. This is necessary because the logger package uses the pflag.Commandline FlagSet. Previously the cobra command FlagSet was auto-parsed when Execute() is called. But, pflag.CommandLine was not parsed. Presently, the flag.CommandLine FlagSet, the cobra command FlagSet have been merged with the pflag.Commandline FlagSet for a more thorough coverage of flag declarations. Finally, before logging/execution, pflag.Parse() is called.
2. The line parsing flag.Commandline with an empty arg slice has been removed. This step is handled in pflag.Parse() (this was required because of https://github.com/kubernetes/kubernetes/issues/17162)
3. Parsing has been moved to the main package. It is good practice to parse in the main package, so that the app package may be imported and the user may implement custom parsing/custom flagset.
4. Does not enforce the `--logtostderr` flag. User configuration is honored. Default to `--logtostderr=false --alsologtostderr=true`. As of klog/v2@v2.40.1 the `--logtostderr=true` option cannot be used alongside other klog flags to write logs in a file or directory. The `--alsologtostderr=true` option can be used alongside other klog flags to write logs to a file or directory. Disabling this flag will disable logging to stderr (while `--logtostderr=false` is set).
Ref: https://github.com/kubernetes/klog/issues/60
5. Included logging package to push messages from standard log stream to klog's log stream.

**Does this PR require any upgrade changes?**:
No.

**If the changes in this PR are manually verified, list down the scenarios covered:**:
1. `--bd-time-out` flag works as expected.
2. `--log_file=/var/log/localpv.log` stores logs in the specified file.
3. `--alsologtostderr=false` disables stderr logging.

### CODE REVIEW

File: cmd/provisioner-localpv/app/helper_blockdevice.go [hunk 1]
```diff
@@ -52,7 +41,8 @@ const (
 
 // WaitForBDTimeoutCounts specifies the duration to wait for BDC to be associated with a BD
 // The duration is the value specified here multiplied by 5
-var WaitForBDTimeoutCounts int
+var WaitForBDTimeoutCounts = pflag.Int("bd-time-out", 12,
```
Comment (akhilerm) [inline cmd/provisioner-localpv/app/helper_blockdevice.go]: Why was this flag moved here. It should be associated with the cobra command right?

File: cmd/provisioner-localpv/main.go [hunk 2]
```diff
@@ -32,9 +43,8 @@ func main() {
 
 // run starts the dynamic provisioner for Local PVs
 func run() error {
-	// Init logging
-	mlogger.InitLogs()
-	defer mlogger.FlushLogs()
+	// Flush logs
+	defer logger.FinishLogging()
```
Comment (akhilerm) [inline cmd/provisioner-localpv/main.go]: Its better to have the initalization and defer close to each other.

**Review by akhilerm — COMMENTED** [review]
@niladrih Provided a few comments

File: cmd/provisioner-localpv/app/helper_blockdevice.go (hunk 1 above)
Comment (niladrih) [reply cmd/provisioner-localpv/app/helper_blockdevice.go]: Moved it back.

File: cmd/provisioner-localpv/main.go (hunk 2 above)
Comment (niladrih) [reply cmd/provisioner-localpv/main.go]: Init and deferred finish functions are now close to each other in the run() function.

File: cmd/provisioner-localpv/main.go [hunk 3]
```diff
@@ -17,12 +17,33 @@ limitations under the License.
...
+	// Declare klog CLI flags
+	klog.InitFlags(flag.CommandLine)
+	// NOTE: As of klog/v2@v2.40.1 the --logtostderr=true option cannot be
+	//       used alongside other klog flags to write logs in a file or
+	//       directory.
+	//       The --alsologtostderr=true option can be used alongside other
+	//       klog flags to write logs to a file or directory. Disabling
+	//       this flag will disable logging to stderr (while
+	//	 --logtostderr=false is set).
+	flag.CommandLine.Set("logtostderr", "false")
```
[the remark this answers is not in the document]
Comment (niladrih) [reply cmd/provisioner-localpv/main.go:42]: ignore

File: cmd/provisioner-localpv/main.go [hunk 4]
```diff
@@ -17,12 +17,35 @@ limitations under the License.
...
+	// NOTE: As of klog/v2@v2.40.1 the --logtostderr=true option cannot be
+	//       used alongside other klog flags to write logs in a file or
+	//       directory.
+	//       The --alsologtostderr=true option can be used alongside other
+	//       klog flags to write logs to a file or directory. Disabling
+	//       this flag will disable logging to stderr (while
+	//	 --logtostderr=false is set).
+	// Ref: https://github.com/kubernetes/klog/issues/60
+	// User flags will be honored at Parse time.
+	flag.CommandLine.Set("logtostderr", "false")
```
[the remark this answers is not in the document]
Comment (niladrih) [reply cmd/provisioner-localpv/main.go:42]: @sonatype-lift ignore

**Review by vharsh — APPROVED** [review]
LGTM, thanks for discussing best practices & minor klog quirks & SO answers. please add them to code comments as applicable so that tomorrow other projects & devs can also benefit from our code & learnings. 👍🏼

**Review by Ab-hishek — APPROVED** [review]
LGTM. Thanks Niladri.

File: cmd/provisioner-localpv/main.go [hunk 5]
```diff
@@ -17,12 +17,32 @@ limitations under the License.
...
+	// Declare klog CLI flags
+	klog.InitFlags(flag.CommandLine)
+	// NOTE: As of klog/v2@v2.40.1 the --logtostderr=true option cannot be
+	//       used alongside other klog flags to write logs in a file or
+	//       directory.
+	//       The --alsologtostderr=true option can be used alongside other
+	//       klog flags to write logs to a file or directory. Disabling
+	//       this flag will disable logging to stderr (while
+	//	 --logtostderr=false is set).
+	flag.CommandLine.Set("logtostderr", "false")
```
[the remark this answers is not in the document]
Comment (niladrih) [reply cmd/provisioner-localpv/main.go:42]: @sonatype-lift ignore

**Pull request merged**